### PR TITLE
Fix attached_geometry updateFromXML

### PR DESCRIPTION
**Pull request by aseth1**
Addresses #1450. 

When `VisualObject` was removed, `Body::updateFromXML` was updated to put `Geometry` in a `geometry` list that was associated with every `Component`.  That code was migrated to `PhysicalFrame` and eventually we associated `attached_geometry` with `Frame`s and that block was not updated. Surprisingly, old model files were still being loaded successfully by being associated with general `components` lists and not with the appropriate `attached_geometry` list of the `Frame`. That has been corrected by simply renaming the node created `Body::updateFromXML` to `attached_geometry` from the original `geometry`.

Should be a short review @aymanhab @klshrinidhi.

### CODE REVIEW

**Comment by aymanhab** [comment]
Thanks @aseth1 the changes look straightforward. The test change only checks for geometry attached to ground,  do you think this's sufficient? One other question, the change from "geometry" to "attached_geometry" had associated change in interpretation (attached have no extra transform while  geometry may have an extra transform). Did you check the case with an older model that had non-trivial transform? Thanks.

**Comment by aseth1** [comment]
> The test change only checks for geometry attached to ground, do you think this's sufficient?

I think it was, but I updated the test so we can verify on any PhysicalFrame by name and I include "ground" for the arm26 model (which as the bulk of the geometry attached to it) and the "spine" in the neck model.

> Did you check the case with an older model that had non-trivial transform?

I did not. I agree we should do a better job verifying that the older `VisibleObject`s with potentially two transforms specified are consistent with their previous interpretation (i.e. #1347), but I think that is beyond the scope of this fix, which aims to put the `Geometry` in the correct list.

File: OpenSim/Tools/Test/testModelCopy.cpp [hunk 1]
```diff
@@ -46,12 +50,14 @@ int main()
     return 0;
 }
 
-void testCopyModel(string fileName)
+void testCopyModel(const string& fileName, const int nbod, 
+    const string& physicalFrameName, const int ngeom)
 {
     const size_t mem0 = getCurrentRSS();
 
     // Automatically finalizes properties by default when loading from file
```
Comment (klshrinidhi) [inline OpenSim/Tools/Test/testModelCopy.cpp]: May remove this line since `finalizeFromProperties()` is called.

**Comment by aymanhab** [comment]
Thanks @aseth1 Let's merge it and will see how old models convert.  LGTM

**Comment by aseth1** [comment]
Thanks @aymanhab and @klshrinidhi.

**Pull request merged**